app/routes/preprints/provider.js

```diff
@@ -10,6 +10,19 @@ import Ember from 'ember';
  */
 export default Ember.Route.extend({
     theme: Ember.inject.service(),
+    session: Ember.inject.service(),
+    currentUser: Ember.inject.service(),
+
+    beforeModel() {
+        if (!this.get('session.isAuthenticated')) {
+            this.replaceWith('not-authenticated');
+        }
+        this.get('currentUser.user').then((user) => {
```

Comment:
Maybe wrap this in `else`? I don't think `replaceWith` stops the flow of execution.

Reply:
Good idea. I ran into that on the other check but didn't hit it with this one for some reason.